src/hooks/useAccess.ts

```diff
@@ -0,0 +1,37 @@
+import { ApiOptions, apiRequest } from './apiRequest';
+
+const ACCESS_BASE_URL = process.env.NEXT_PUBLIC_ACCESS_BASE_URL;
+
+/**
+ * Interface representing the structure of an API request bundle
+ * @interface RequestBundle
+ */
+export interface RequestBundle {
+  endpoint: string;
+  options: ApiOptions;
+}
+
+/**
+ * @returns Object containing request method for making API calls
+ */
+export const useAccess = () => {
+  /**
+   * @template T Type of the expected response data
+   * @param requestBundle - Bundle containing endpoint and request options
+   * @returns {Promise<T>} Promise resolving to the response data
+   * @throws {Error} When the API request fails
+   */
+  const request = async <T>(requestBundle: RequestBundle): Promise<T> => {
+    try {
+      const accessUri = `${ACCESS_BASE_URL}${requestBundle.endpoint}`;
+      return await apiRequest<T>(accessUri, {
+        ...requestBundle.options,
+      });
+    } catch (error) {
+      console.error('Error in request:', error);
+      throw error;
+    }
+  };
```

Comment:
The try-catch block here is not necessary because if `apiRequest` throws an error if its `!response` condition is true, the promise will contain the error that's going to be returned to the calling function. Note that this can be confirmed with the negative passing unit test that remains unchanged. The only use for the catch block appears to be used to print the error to the console which we don't want.

```suggestion
  const request = async <T>(requestBundle: RequestBundle): Promise<T> => {
    const accessUri = `${ACCESS_BASE_URL}${requestBundle.endpoint}`;

    return await apiRequest<T>(accessUri, {
      ...requestBundle.options,
    });
  };
```

Reply:
Note that once we build the `FlashMessage` component, if we need to update this somehow so that it sets the `FlashMessage` state that triggers a render, we can make that change as needed (that is if it'll happen here or in the calling function).